### DISCUSSION TITLE
Catalyst is about the user, tell me what features YOU want in Catalyst.

### DISCUSSION THREAD

**Discussion by jdev082** [CatalystDevOrg/Catalyst#330] [Announcements]
Right now, the top feature in the vote is a built in HTML editor. What side features and improvements do **you** want to see in Catalyst?

**Comment by redstone-dev** [comment]
It would be cool if you could do a "bookmarks menu" instead of a bookmarks bar, which would sort of look like Windows 8's Start Screen.
![image](https://user-images.githubusercontent.com/31410740/214728593-a6570e4d-7e54-4a12-8ee0-4dd516e0907c.png)

**Reply by jdev082** [reply]
bookmark tiles :heavy_check_mark:, coming soon! Unfortunately not in v3.4 though, maybe v3.5
also, grids are hard

**Reply by redstone-dev** [reply]
I can maybe implement it.

**Reply by jdev082** [reply]
@redstone-scratch added in v3.4

**Comment by imdevarsh** [comment]
I would use catalyst if it had some good unique features.
Such as extension support, if it is added then it would be awesome!
Also there could be a Dev mode inside of Catalyst which could start a web server and let you develop HTML, SvelteKit and more inside of your browser using the Monaco editor. The difference between this and cloud-based services would be that it runs on your local machine. Sandboxing might be an issue though.

**Reply by jdev082** [reply]
These features are getting complicated for single person development. If you can, any help on the built in HTML editor would be appreciated.

**Reply by jdev082** [reply]
Unfortunately extensions aren't happening though.

**Comment by TechStudent10** [comment]
Theming engine: add and save custom CSS for websites.

**Reply by jdev082** [reply]
This would be very difficult to implement, you must assist in implementation otherwise this feature will not be added.

**Reply by redstone-dev** [reply]
just inject a `<style>` tag and save styles in compressed .css files

to use `fs` you can disable `contextIsolation` and enable `nodeIntegration`

**Reply by jdev082** [reply]
> just inject a `<style>` tag and save styles in compressed .css files
> 
> to use `fs` you can disable `contextIsolation` and enable `nodeIntegration`

I don't recall the renderer ever having access to npm modules

**Reply by redstone-dev** [reply]
well it does, you just need to change those settings in the `createWindow` function

so basically just add
```js
createWindow({
// ...
    contextIsolation: false,
    nodeIntegration: true
// ...
})
```

(the `createWindow()` function is just there to show you where to put the items)

**Reply by jdev082** [reply]
> well it does, you just need to change those settings in the `createWindow` function

Directly in a JS file loaded in the HTML?

**Reply by TechStudent10** [reply]
> > just inject a `<style>` tag and save styles in compressed .css files
> > to use `fs` you can disable `contextIsolation` and enable `nodeIntegration`
> 
> I don't recall the renderer ever having access to npm modules

It **DOES**, it's just not recommended.

**Reply by jdev082** [reply]
> > > just inject a `<style>` tag and save styles in compressed .css files
> > > to use `fs` you can disable `contextIsolation` and enable `nodeIntegration`
> > 
> > I don't recall the renderer ever having access to npm modules
> 
> It **DOES**, it's just not recommended.

Can i deny the WebView access? It can be a huge security issue otherwise.

**Reply by TechStudent10** [reply]
> well it does, you just need to change those settings in the `createWindow` function
> 
> so basically just add
> 
> ```js
> createWindow({
> // ...
> contextIsolation: false,
> nodeIntegration: true
> // ...
> })
> ```

This is an insecure way of doing it. Instead, you should use the IPC system.

**Reply by jdev082** [reply]
> > well it does, you just need to change those settings in the `createWindow` function
> > 
> > so basically just add
> > 
> > ```js
> > createWindow({
> > // ...
> > contextIsolation: false,
> > nodeIntegration: true
> > // ...
> > })
> > ```
> 
> This is an insecure way of doing it. Instead, you should use the IPC system. 

Not once have i ever gotten IPC to work. If you want security, you do it.

**Reply by redstone-dev** [reply]
> > > well it does, you just need to change those settings in the `createWindow` function
> > > so basically just add
> > > ```js
> > > createWindow({
> > > // ...
> > > contextIsolation: false,
> > > nodeIntegration: true
> > > // ...
> > > })
> > > ```
> > 
> > 
> > This is an insecure way of doing it. Instead, you should use the IPC system.
> 
> Not once have i ever gotten IPC to work. If you want security, you do it.

[Please just read documentation for once.](https://www.electronjs.org/docs/latest/tutorial/ipc)

**Reply by TechStudent10** [reply]
> > > > just inject a `<style>` tag and save styles in compressed .css files
> > > > to use `fs` you can disable `contextIsolation` and enable `nodeIntegration`
> > > 
> > > 
> > > I don't recall the renderer ever having access to npm modules
> > 
> > 
> > It **DOES**, it's just not recommended.
> 
> Can i deny the WebView access? It can be a huge security issue otherwise.

Just don't use this approach entirely.

Rather, use the IPC system.

```js
const {ipcMain} = require("electron")

// Later on...

// This should be in `app.whenReady().then()

ipcMain.handle("do-something", (event) => {
    return "this"
})
```

**Reply by jdev082** [reply]
> > > > well it does, you just need to change those settings in the `createWindow` function
> > > > so basically just add
> > > > ```js
> > > > createWindow({
> > > > // ...
> > > > contextIsolation: false,
> > > > nodeIntegration: true
> > > > // ...
> > > > })
> > > > ```
> > > 
> > > 
> > > This is an insecure way of doing it. Instead, you should use the IPC system.
> > 
> > Not once have i ever gotten IPC to work. If you want security, you do it.
> 
> [Please just read documentation for once.](https://www.electronjs.org/docs/latest/tutorial/ipc)

What do you think i went off of trying get it to work

**Reply by TechStudent10** [reply]
In my own app, I got it to work.

**Reply by TechStudent10** [reply]
The app is at https://github.com/ProtonDesigner/core

**Comment by PPPDUD** [comment]
Please add Ogle Home as a homepage option, and maybe discontinue Catalyst Mobile so that we can be friends and stop competing.
